src/Functions/translate.cpp

```diff
@@ -32,17 +32,31 @@ struct TranslateImpl
         const std::string & map_from,
         const std::string & map_to)
     {
-        if (map_from.size() != map_to.size())
-            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Second and third arguments must be the same length");
-
         iota(map.data(), map.size(), UInt8(0));
 
-        for (size_t i = 0; i < map_from.size(); ++i)
+        size_t min_size = std::min(map_from.size(), map_to.size());
```

Comment:
The case that the third argument is longer than the second argument is weird, it should throw an exception: `select translate('abc', 'ab', 'abc')`.  @yariks5s ?

Reply:
https://github.com/ClickHouse/ClickHouse/pull/72893